cmd/system-probe/config/config.go

```diff
@@ -156,6 +157,17 @@ func load() (*Config, error) {
 		cfg.Set("log_file", c.LogFile)
 	}
 
+	// This code block handles the backward compatibility logic for the enable_http_monitoring configuration value
+	deprecatedEnableHttpMonitoringKey := key(netNS, "enable_http_monitoring")
+	if cfg.IsSet(deprecatedEnableHttpMonitoringKey) {
+		enableHttpMonitoringKey := key(smNS, "enable_http_monitoring")
+		log.Infof("%q is deprecated, use %q instead",
+			deprecatedEnableHttpMonitoringKey, enableHttpMonitoringKey)
+		if !cfg.IsSet(enableHttpMonitoringKey) {
+			cfg.Set(enableHttpMonitoringKey, cfg.GetBool(deprecatedEnableHttpMonitoringKey))
+		}
+	}
+
```

Comment:
nit: can be done in a different PR
as we know we're going to have more deprecated configuration entries, I do suggest to create a dedicated function to store all of them